extensions/resource-deployment/src/test/services/notebookService.test.ts

```diff
@@ -3,82 +3,108 @@
+	const notebookDarwin = 'test-notebook-darwin.ipynb';
+	const notebookLinux = 'test-notebook-linux.ipynb';
+	const storagePath = __dirname;
+	let mockPlatformService: TypeMoq.IMock<IPlatformService>, notebookService: NotebookService;
+
+	beforeEach('NotebookService Setup', () => {
+		mockPlatformService = TypeMoq.Mock.ofType<IPlatformService>();//PlatformService, TypeMoq.MockBehavior.Loose
+		notebookService = new NotebookService(mockPlatformService.object, extensionPath);
+	});
+
+	afterEach('NotebookService cleanup', () => {
+		sinon.restore();
+		mockPlatformService.reset();
+	});
 
-describe('Notebook Service Tests', function (): void {
+	it('getNotebook with string parameter', async () => {
+		mockPlatformService.setup(x => x.fileExists(notebookInput)).returns(async () => true);
+		mockPlatformService.setup(x => x.readTextFile(notebookInput)).returns(async () => sourceNotebookContent);
+		let returnValue = await notebookService.getNotebook(notebookInput);
+		assert.deepEqual(returnValue, JSON.parse(sourceNotebookContent), 'returned notebook does not match expected value');
+	});
 
-	it('getNotebook with string parameter', () => {
-		const mockPlatformService = TypeMoq.Mock.ofType<IPlatformService>();
-		const notebookService = new NotebookService(mockPlatformService.object, '');
-		const notebookInput = 'test-notebook.ipynb';
+	it('getNotebookPath with string parameter', () => {
 		mockPlatformService.setup((service) => service.platform()).returns(() => { return 'win32'; });
 		let returnValue = notebookService.getNotebookPath(notebookInput);
-		assert.equal(returnValue, notebookInput, 'returned notebook name does not match expected value');
+		assert.strictEqual(returnValue, notebookInput, 'returned notebook name does not match expected value');
 		mockPlatformService.verify((service) => service.platform(), TypeMoq.Times.never());
 
 		mockPlatformService.reset();
 		mockPlatformService.setup((service) => service.platform()).returns(() => { return 'win32'; });
 		returnValue = notebookService.getNotebookPath('');
-		assert.equal(returnValue, '', 'returned notebook name does not match expected value is not an empty string');
+		assert.strictEqual(returnValue, '', 'returned notebook name does not match expected value is not an empty string');
 		mockPlatformService.verify((service) => service.platform(), TypeMoq.Times.never());
 	});
 
 	it('getNotebook with NotebookInfo parameter', () => {
-		const mockPlatformService = TypeMoq.Mock.ofType<IPlatformService>();
-		const notebookService = new NotebookService(mockPlatformService.object, '');
-		const notebookWin32 = 'test-notebook-win32.ipynb';
-		const notebookDarwin = 'test-notebook-darwin.ipynb';
-		const notebookLinux = 'test-notebook-linux.ipynb';
-
 		const notebookInput: NotebookPathInfo = {
 			darwin: notebookDarwin,
 			win32: notebookWin32,
 			linux: notebookLinux
 		};
 		mockPlatformService.setup((service) => service.platform()).returns(() => { return 'win32'; });
```

Comment:
From @Charles-Gagnon on #13183
Please use should assertions - the output error messages are typically a lot better

Again - I'm really not a fan of forcing platforms like this. This one isn't really testing actual logic so it's ok I guess but I'd rather we just avoid all this kind of stuff in general unless there's a really good reason not to. Getting a couple extra lines of code coverage isn't really beneficial for the added test complexity and possible side effects of having a test pretend it's running on a certain platform that it isn't actually running on.

This could just be simplified to having all platforms returning the same notebook - that way you just need the one check.

Reply:
Agreed. However, that is pre-existing code. Let's clean that up later.